src/shared/sleep-config.c

```diff
@@ -108,77 +111,192 @@ int parse_sleep_config(SleepConfig **ret_sleep_config) {
         if (r < 0)
-               return log_debug_errno(r, "Failed to read /sys/class/power_supply/BAT0/capacity: %m");
+                return log_device_debug_errno(dev, r, "Failed to read battery capacity: %m");
 
-        r = safe_atoi(bat_cap, &battery_capacity);
+        r = safe_atoi(power_supply_capacity, &battery_capacity);
         if (r < 0)
-               return log_debug_errno(r, "Failed to parse battery capacity: %m");
+                return log_debug_errno(r, "Failed to parse battery capacity: %m");
 
         if (battery_capacity < 0 || battery_capacity > 100)
-               return log_debug_errno(SYNTHETIC_ERRNO(ERANGE), "Invalid battery capacity");
+                return log_debug_errno(SYNTHETIC_ERRNO(ERANGE), "Invalid battery capacity");
 
         return battery_capacity;
 }
 
+/* If battery percentage capacity is less than equal to 5% return success */
+int battery_is_low(void) {
+        _cleanup_(sd_device_enumerator_unrefp) sd_device_enumerator *e = NULL;
+        sd_device *dev;
+        bool c = false;
+        int r;
+
+         /* We have not used battery capacity_level since value is set to full
+         * or Normal in case acpi is not working properly. In case of no battery
+         * 0 will be returned and system will be suspended for 1st cycle then hibernated */
+
+        r = get_sd_device_battery(&e);
+        if (r < 0)
+                return log_error_errno(r, "Failed to get available battery: %m");
+
+        FOREACH_DEVICE(e, dev) {
+                r = read_battery_capacity_percentage(dev);
+                if (r < 0) {
+                        log_debug_errno(r, "Failed to get battery capacity: %m");
+                        continue;
+                }
+                if (r > 5)
+                        /* true only if battery capacity is above 5 */
+                        c = true;
+        }
+        /* if all batteries are below 5 then c will remain false */
+        if (!c)
+                return 1;
+
+        return 0;
+}
+
+/* get sum of all batteries current charge level */
+int get_total_battery_capacity(void) {
+        _cleanup_(sd_device_enumerator_unrefp) sd_device_enumerator *e = NULL;
+        sd_device *dev;
+        int total_battery_capacity, r;
+
+        r = get_sd_device_battery(&e);
+        if (r < 0)
+                return log_error_errno(r, "Failed to get available battery: %m");
+
+        FOREACH_DEVICE(e, dev) {
+                r = read_battery_capacity_percentage(dev);
+                if (r < 0) {
+                        log_debug_errno(r, "Failed to get battery capacity: %m");
+                        continue;
+                }
+                total_battery_capacity = total_battery_capacity + r;
+        }
+
+        return total_battery_capacity;
+}
+
+/* Store current capacity of each battery before suspension and timestamp */
+int hashmap_battery_name_with_battery_capacity(Hashmap **ret_current_capacity) {
```

Comment:
If a function returns only one object, then we usually simply name it as `ret`.

Reply:
yes. fixed